docs/pages/contributing/documentation/how-to-contribute.mdx

```diff
@@ -1,168 +1,306 @@
+    ```
+
+1. Start the development server by running the following commands:
+    
+    ```code
+    $ yarn install
+    $ yarn dev
+    ```
+
+1. Open a web browser and navigate to the documentation using the URL `localhost:3000/docs`. 
+
+1. Validate the Markdown syntax by running the following commends:
+
+    ```code
+    # Check your Markdown syntax
+    $ yarn markdown-lint
+    # markdown-lint, but also checks that external links work
+    $ yarn markdown-lint-external-links
+    ```
+
+    For more information about using `yarn` commands to work with Teleport documentation, 
+    see the [README](https://github.com/gravitational/docs).
+
+## Step 4/5. Create a pull request
+
+If you're an internal contributor with permission to access the [Teleport repository](https://github.com/gravitational/teleport), you can commit changes and push branches directly to the repository.
+If you're an external contributor, you can commit changes to your private branch and push private branches to a fork of the [Teleport repository](https://github.com/gravitational/teleport).
+
+### Internal contributors
+
+If you work at Teleport:
+
+1. Add files and commit changes to your local branch periodically with commands similar to the following:
+
+    ```code
+    $ git add <new-file-name>                      # Add a specific new file to be committed
+    $ git add -A                                   # Add all changed files to the list of files to be committed
+    $ git commit -am "Fix or feature description"  # Add and commit changes with a comment
+    $ git commit -m                                # Commit changes already added
+    ```
+
+1. Push your branch to `gravitational/teleport` with a command similar to the following:
+    
+    ```code
+    $ git push origin my-branch
+    ```
+
+1. Open the [Teleport repository](https://github.com/gravitational/teleport), select your branch as the branch to merge into main, then click **New pull request**.
+
+### External contributors
+
+If you are an external contributor, you must commit your changes to your branch and push your branch to a fork of the `gravitational/teleport` repository.
+
+To push changes as an external contributor:
+
+1. Verify that you have an SSH key pair and have stored your public key in GitHub.
+
+1. Open the [Teleport repository](https://github.com/gravitational/teleport), click **Fork**, then select **Create a new fork**.
+
+1. Verify the owner and repository name.
+
+1. Click **Create fork**.
+    
+    Alternatively, you can create a fork from the command-line by running a command similar to the following:
+    
+    ```code
+    $ git remote add fork ssh://git@github.com/my-user/teleport
+    ```
+
+    In this command, `my-user` represents your GitHub user name and `teleport` is the name of your Teleport repository fork.
+
+1. Pull all of the changes from the remote repository into your fork by running the following command:
+    
+    ```code
+    $ git rebase origin/master
```

Comment:
This doesn't pull anything from the remote, it just rebases on whatever you have locally. I'd recommend a `git fetch origin` first to make sure your local copy is up to date.